micarraylib/utils.py

```diff
@@ -63,19 +65,19 @@ def _get_audio_numpy(
         )
     if fmt_in == "B" and fmt_out == "A":
         raise ValueError("B to A conversion currently not supported")
-    if fmt_out == "B" and N != None and (N + 1) ** 2 > len(clip_names):
-        raise ValueError(
-            "(N+1)^2 should be less than or equal to the number of channels being combined but (N+1)^2 is {} and len(clip_names) is {}".format(
-                (N + 1) ** 2, len(clip_names)
-            )
-        )
     if fmt_in == "A" and fmt_out == "B" and capsule_coords == None:
         raise ValueError(
             "To convert between A and B format you must specify capsule coordinates"
         )
     audio_data = [all_dataset_clip_names[ac].audio for ac in clip_names]
     audio_array, audio_fs = list(map(list, zip(*audio_data)))
     audio_array = np.squeeze(np.array(audio_array))
+    if fmt_out == "B" and N != None and (N + 1) ** 2 > len(audio_array):
+        raise ValueError(
+            "(N+1)^2 should be less than or equal to the number of channels being combined but (N+1)^2 is {} and len(audio_array) is {}".format(
```

Comment:
Thanks for solving this issue. I think that your solution is the best one we can do with the current implementation. After looking through this PR I think it would be better if this `ValueError` message reads `"(N+1)^2 should be less than or equal to the number of capsules being converted to B format, but (N+1)^2 is {} and the number of capsules is" . . .`.

Can you please make that change so that it's included as part of this PR?

Reply:
That's more accurate, good catch